File: Scripts/perceptron.py
import random
class Perceptron:
    def __init__(self, numberOfWeights):
        self.weights= []
        
        for i in range(numberOfWeights):
            self.weights.append(0)
        self.numberOfWeights = numberOfWeights
        for i in range(numberOfWeights):
            self.weights[i] = random.uniform(-1.1,1.1)
        print("weights:")
        print(self.weights)
    
    #Length of inputs must be the same as weights
    def createSum(self, inputs):
        sum = 0
        for i in range(len(inputs)):
            sum += (inputs[i] * self.weights[i])
        
        output = 0
        if sum >=0 :
            output = 1
        else:
            output=0
        return output
        

File: Assets/Scripts/NeuralNetwork/Perceptron.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Perceptron
{
    private int numberOfWeights;
    private float[] weights;
    public Perceptron(int numberOfWeights){
        this.numberOfWeights = numberOfWeights;
        this.weights = createWeights();
    }
    public Perceptron(int numberOfWeights, float[] weights){
        this.numberOfWeights = numberOfWeights;
        this.weights = weights;
    }
    public float[] createWeights(){
        float[] weights = new float[numberOfWeights];
        for(int i=0; i<numberOfWeights; i++){
            weights[i] = Random.Range(-1f,1.1f);
        }
        return weights;
    }
    public float createSum(float[] inputs){
        float sum = 0;
        for(int i=0; i<numberOfWeights; i++){
            sum += (inputs[i] * weights[i]);
        }
        return sum;
    }
    public float[] getWeights(){
        return weights;
    }
}
